CMakeLists.txt

```diff
@@ -221,6 +216,17 @@ if (NOT CMAKE_CXX_FLAGS)
   # Increase maximum number of template instantiations, for all that template-
   # heavy code.
   set(EXTRA_FLAGS "${EXTRA_FLAGS} -ftemplate-backtrace-limit=0")
+  # Ninja doesn't colorize compiler diagnostics by default.
+  if (CMAKE_GENERATOR STREQUAL "Ninja")
+    set(EXTRA_FLAGS "${EXTRA_FLAGS} -fdiagnostics-color")
+  endif ()
+  # For clang, relocate debug/file/macro paths to a common prefix.
+  # This massively improves recompilation speed with CCache when working from
+  # different worktrees.
+  if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
This option is not specific to Clang. You can also shorten the explanation.

Reply:
Done and rebased. I thought this was clang only.